Assets/Scripts/PlayerCharacter.cs

```diff
@@ -57,87 +68,127 @@ public class PlayerCharacter : MonoBehaviour
 
     // Update is called once per frame
     void Update() {
-        if (Time.realtimeSinceStartup - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        
+        if (deathTimer>timeToDeath + deathBuffer || deathTimer == 0 || !isDead)
         {
             if (Input.GetButtonDown("Jump") && jumpCooldown == 0)
             {
                 Jump();
-                //allows player to jump and checks to make sure character can't jump when dead
-            }
-            deathTimer = 0;
-        }
-        if (isInDeath)
-        {
-            anim.SetBool("IsDead", true);
-            if (Time.realtimeSinceStartup - deathTimer >= timeToDeath)
-            {
-                Respawn();
-                //if death timer has gone up and matches time to death, it will enter the next stage of respawn.
-                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
             }
-            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
+            DeathTimerZero();
```

Comment:
I'd prefer method names to be verbs. Tell me what action this takes in its name.